src/movies.js

```diff
@@ -1,16 +1,109 @@
-// Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors (array) {
+  const directorArray = array.map(movie => {
+    return movie.director
+  })
+  return directorArray
+}
+// console.log(getAllDirectors(movies));
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+// // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+
+function howManyMovies (array) {
+  const directorSpielberg = array.filter((movie) => {
+    if (movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')) {
+      return movie
+    }
+  })
+  const onlyTitle = directorSpielberg.map(movie => movie.title)
+  return onlyTitle.length
+}
+// console.log(howManyMovies(movies))
 
-// Iteration 3: All rates average - Get the average of all rates with 2 decimals
+// // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage (array) {
+  const sumRatingCalc = array.reduce((total, movie) => {
+    return (total + (movie.rate || 0)) // Does this make sense??? [6, , ] should average out to 6 not 2
```

Comment:
This can be another solution: 
` const sumRatingCalc = array.reduce((total, movie) => {`
    `if (movie.rate) {`
    `  return total + movie.rate`
    `} else {`
  `    return total`
   ` }`
`  }, 0)`